ts/main.ts

```diff
@@ -151,11 +161,44 @@ $list.addEventListener('click', (event: Event) => {
 
   data.editing = entry;
 
-  $photo.setAttribute('src', entry.url);
+  $formPhoto.setAttribute('src', entry.url);
   $photoURL.value = entry.url;
   $formTitle.value = entry.title;
   $formNotes.value = entry.notes;
   $newOrEditing.textContent = 'Edit Entry';
+  $deleteButton.className = 'delete';
+});
+
+$deleteButton.addEventListener('click', () => {
+  $confirmationScreen.showModal();
+});
+
+$cancelDelete.addEventListener('click', () => {
+  $confirmationScreen.close();
+});
+
+$confirmDelete.addEventListener('click', () => {
+  if (!data.editing) throw new Error('attempted to delete while not editing');
+
+  const index = data.entries.findIndex(
+    (e) => e.entryId === data.editing?.entryId,
+  );
+
+  data.entries.splice(index, 1);
+
+  const $deletedEntry = document.querySelector(
+    `[data-entry-id = '${data.editing.entryId}']`,
+  );
+  if (!$deletedEntry) throw new Error('$deletedEntry query failed');
+
+  $deletedEntry.remove();
+  checkNoEntries();
+
+  data.editing = null;
+  $confirmationScreen.close();
+  data.view = 'entries';
```

Comment:
This will already get updated inside of `viewSwap`, so no need to update it here as well